bundles/org.openhab.ui/web/src/components/widgets/layout/oh-canvas-layout.vue

```diff
@@ -0,0 +1,260 @@
+    background-color: var(--f7-page-bg-color); // theme bg (for fullscreen)
+    transform-origin: top;
+  }
+
+  .oh-canvas-background {
+    height: 100%;
+    width: 100%;
+    object-fit: contain;
+  }
+
+  .fullscreen-icon {
+    position: absolute;
+    top: 2px;
+    right: 2px;
+    z-index: 1000;
+  }
+}
+</style>
+
+<script>
+import mixin from '../widget-mixin'
+import OhCanvasItem from './oh-canvas-item'
+import { OhCanvasLayoutDefinition } from '@/assets/definitions/widgets/layout'
+
+export default {
+  mixins: [mixin],
+  widget: OhCanvasLayoutDefinition,
+  components: {
+    OhCanvasItem: OhCanvasItem
+  },
+  data () {
+    return {
+      layout: [],
+      screenWidth: Number,
+      screenHeight: Number,
+      fullscreen: this.$fullscreen.getState(),
+      navbarHidden: false,
+      style: {
+        width: Number,
+        height: Number,
+        scale: 1.0
+      },
+      grid: {
+        pitch: Number,
+        enable: false
+      }
+    }
+  },
+  created () {
+    if (this.config.layoutType === 'fixed' && this.config.fixedType === 'canvas') {
+      this.style.width = this.screenWidth = this.config.screenWidth || 1280
+      this.style.height = this.screenHeight = this.config.screenHeight || 720
+      this.grid.pitch = this.config.grid || 5
+      this.grid.enable = this.config.gridEnable || false
+
+      if (!this.context.editmode) {
+        window.addEventListener('resize', this.setDimensions)
+      } else {
+        window.addEventListener('keydown', this.doCommand)
+        console.log('Adding keydown window listener')
+      }
+    }
+
+    this.canvasLayoutStyle()
+    this.computeLayout()
+  },
+  mounted () {
+    // Chrome reports a wrong size in fullscreen, store initial resolution and use non-dynamically.
+    this.windowWidth = window.screen.width
+    this.windowHeight = window.screen.height
+  },
+  destroyed () {
+    window.removeEventListener('keydown', this.doCommand)
+  },
+  methods: {
+    isRetina () {
+      return window.devicePixelRatio > 1
+    },
+    getCurrentScreenResolution () {
+      return (
+        'Layout Size: ' +
+        this.screenWidth +
+        ' x ' +
+        this.screenHeight +
+        ' (Current Screen: ' +
+        this.windowWidth +
+        ' x ' +
+        this.windowHeight +
+        ')'
+      )
+    },
+    addItem () {
+      this.context.component.slots['canvas'].push({
+        component: 'oh-canvas-item',
+        config: { x: 20, y: 20, h: 150, w: 200 },
+        slots: { default: [] }
+      })
+      this.computeLayout()
+    },
+    toggleFullscreen () {
+      this.$fullscreen.toggle(this.$refs.ohCanvasLayout, {
+        wrap: false,
+        callback: (fullscreen) => {
+          this.canvasLayoutStyle()
+          this.fullscreen = fullscreen
+        }
+      })
+    },
+    toggleGrid () {
+      this.context.component.config.gridEnable = this.grid.enable = !this.grid.enable
+    },
+    canvasLayoutStyle: function () {
```

Comment:
Any reason it isn't `canvasLayoutStyle () {` ?

Reply:
Changed, including a few others lingering...